wpilibj/src/main/java/edu/wpi/first/wpilibj/buttons/XboxControllerButton.java

```diff
@@ -0,0 +1,151 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2008-2017 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpilibj.buttons;
+
+import edu.wpi.first.wpilibj.XboxController;
+
+/**
+ * A {@link Button} that gets its state from an {@link XboxController}.
+ */
+public class XboxControllerButton extends Button {
+  /**
+   * Represents a digital button on an XboxController.
+   */
+  private enum Button {
```

Comment:
Instead of duplicating this enum, why not make `XboxController.Button` public and reuse that?